airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/conditional_page_paginator.py

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, Optional
+
+import requests
+from airbyte_cdk.sources.declarative.decoders.decoder import Decoder
+from airbyte_cdk.sources.declarative.interpolation.interpolated_boolean import InterpolatedBoolean
+from airbyte_cdk.sources.declarative.states.dict_state import DictState
+
+
+class ConditionalPagePaginator:
```

Comment:
The name page paginator seems redundant to me, but I also think it's important to delineate that this increments pages, what do you think? Could also just go with `ConditionalPaginator` for simplicity